src/plugins/settings/setting.ts

```diff
@@ -13,20 +13,14 @@ import {WARNING_MSG} from '../../utils/warning-msg';
  */
 export abstract class UIPSetting extends ESLBaseElement {
   static is = 'uip-setting';
-  /** Event fired when setting's value is changed. */
-  static changeEvent = 'change';
 
   /** [Target's]{@link target} attribute which is changed by setting. */
   @attr() public attribute: string;
   /** Target to which setting's changes are attached. */
   @attr() public target: string;
 
-  public get settingContainer(): HTMLElement | null {
-    return this.closest(UIPSettings.is);
-  }
-
-  public get settings() {
-    return this.settingContainer as UIPSettings;
+  public get $settings() {
+    return this.closest(UIPSettings.is) as UIPSettings;
```

Comment:
cosmetic: guess we should use memoize decorator here

Reply:
good catch, thank you, fixed!